File: frontend/src/types/finding.ts
export type FindingSeverity = 'info' | 'low' | 'medium' | 'high' | 'critical';
export type FindingStatus = 'open' | 'accepted' | 'false_positive' | 'fixed';

export interface Finding {
  id: number;
  target_id: number;
  asset_id?: number | null;
  url_id?: number | null;
  title: string;
  description?: string;
  severity: FindingSeverity;
  category?: string;
  source_tool?: string;
  evidence?: string;
  recommendation?: string;
  status: FindingStatus;
  triage_note?: string;
  triaged_at?: string | null;
  triaged_by_user_id?: number | null;
  fingerprint?: string;
  first_seen?: string;
  last_seen?: string;
  created_at?: string;
  updated_at?: string;
}

export interface FindingsResponse {
  status: string;
  data: Finding[];
  count?: number;
  total?: number;
  total_count?: number;
  page?: number;
}

export interface FindingStats {
  total: number;
  open: number;
  accepted: number;
  false_positive: number;
  fixed: number;
  by_severity: Record<FindingSeverity, number> & Record<string, number>;
  by_status: Record<FindingStatus, number> & Record<string, number>;
  by_source: Record<string, number>;
  by_category: Record<string, number>;
}

export interface FindingStatsResponse {
  status: string;
  data: FindingStats;
}

File: backend/internal/api/dto/finding.go
package dto

import (
	"encoding/json"
	"time"
)

type FindingResponse struct {
	ID              uint            `json:"id"`
	TargetID        uint            `json:"target_id"`
	AssetID         *uint           `json:"asset_id"`
	URLID           *uint           `json:"url_id"`
	Title           string          `json:"title"`
	Description     string          `json:"description"`
	Severity        string          `json:"severity"`
	Category        string          `json:"category"`
	SourceTool      string          `json:"source_tool"`
	Evidence        string          `json:"evidence"`
	EvidenceJSON    json.RawMessage `json:"evidence_json"`
	Recommendation  string          `json:"recommendation"`
	Status          string          `json:"status"`
	TriageNote      string          `json:"triage_note"`
	TriagedAt       *time.Time      `json:"triaged_at"`
	TriagedByUserID *uint           `json:"triaged_by_user_id"`
	Fingerprint     string          `json:"fingerprint"`
	FirstSeen       time.Time       `json:"first_seen"`
	LastSeen        time.Time       `json:"last_seen"`
	CreatedAt       time.Time       `json:"created_at"`
	UpdatedAt       time.Time       `json:"updated_at"`
}

type UpdateFindingStatusRequest struct {
	Status     string `json:"status"`
	TriageNote string `json:"triage_note"`
}

type FindingStatsResponse struct {
	Total         int64            `json:"total"`
	Open          int64            `json:"open"`
	Accepted      int64            `json:"accepted"`
	FalsePositive int64            `json:"false_positive"`
	Fixed         int64            `json:"fixed"`
	BySeverity    map[string]int64 `json:"by_severity"`
	ByStatus      map[string]int64 `json:"by_status"`
	BySource      map[string]int64 `json:"by_source"`
	ByCategory    map[string]int64 `json:"by_category"`
}
